src/platform/datapath_raw.c

```diff
@@ -17,6 +17,56 @@
 #pragma warning(disable:4116) // unnamed type definition in parentheses
 #pragma warning(disable:4100) // unreferenced formal parameter
 
+CXPLAT_THREAD_CALLBACK(CxPlatRouteResolutionWorkerThread, Context);
+
+_IRQL_requires_max_(PASSIVE_LEVEL)
+QUIC_STATUS
+CxPlatDataPathRouteWorkerInitialize(
+    _Inout_ CXPLAT_DATAPATH* DataPath
+    )
+{
+    QUIC_STATUS Status;
+    CXPLAT_ROUTE_RESOLUTION_WORKER* Worker =
+        CXPLAT_ALLOC_NONPAGED(
+            sizeof(CXPLAT_ROUTE_RESOLUTION_WORKER), QUIC_POOL_ROUTE_RESOLUTION_WORKER);
+    if (Worker == NULL) {
+        QuicTraceEvent(
+            AllocFailure,
+            "Allocation of '%s' failed. (%llu bytes)",
+            "CXPLAT_DATAPATH",
+            sizeof(CXPLAT_ROUTE_RESOLUTION_WORKER));
+        Status = QUIC_STATUS_OUT_OF_MEMORY;
+        goto Error;
+    }
+
+    CxPlatEventInitialize(&Worker->Ready, FALSE, FALSE);
+    CXPLAT_THREAD_CONFIG ThreadConfig = {
+        CXPLAT_THREAD_FLAG_NONE,
+        0,
+        "RouteResolutionWorkerThread",
+        CxPlatRouteResolutionWorkerThread,
+        Worker
+    };
+
+    Status = CxPlatThreadCreate(&ThreadConfig, &Worker->Thread);
+    if (QUIC_FAILED(Status)) {
+        QuicTraceEvent(
+            LibraryErrorStatus,
+            "[ lib] ERROR, %u, %s.",
+            Status,
+            "CxPlatThreadCreate");
+        goto Error;
+    }
+
+    DataPath->RouteResolutionWorker = Worker;
+
+Error:
+    if (QUIC_FAILED(Status) && Worker != NULL) {
+        CXPLAT_FREE(Worker, QUIC_POOL_ROUTE_RESOLUTION_WORKER);
```

Comment:
Leaking `Worker->Ready` in case of `CxPlatThreadCreate` failure.

Reply:
Fixed